src/writeData/components/AddPluginToConfiguration.tsx

```diff
@@ -30,84 +41,121 @@ interface AddPluginToConfigurationCTAProps {
   >
 }
 
-export const AddPluginToConfigurationCTA: FC<AddPluginToConfigurationCTAProps> = ({
-  contentID,
-  history,
-  orgID,
-  thumbnail,
-  pageContent,
-}) => (
-  <>
-    <div className="write-data--details">
-      <Grid>
-        <Grid.Row>
-          <Grid.Column
-            widthXS={Columns.Twelve}
-            widthMD={Columns.Six}
-            widthLG={Columns.Three}
-          >
-            <div className="write-data--details-thumbnail">{thumbnail}</div>
-            <Dropdown
-              button={(active, onClick) => (
-                <Dropdown.Button
-                  active={active}
-                  color={ComponentColor.Primary}
-                  style={{width: '100%'}}
-                  onClick={onClick}
-                >
-                  Use this plugin
-                </Dropdown.Button>
-              )}
-              className="use-plugin--telegraf-configuration"
-              menu={onCollapse => (
-                <Dropdown.Menu onCollapse={onCollapse}>
-                  <Dropdown.Item
-                    key="Create-new-configuration-telegraf-plugin"
-                    value="Create a new configuration"
-                    onClick={() => {
-                      history.push(
-                        `/${ORGS}/${orgID}/load-data/${TELEGRAF_PLUGINS}/${contentID}/new`
-                      )
-                    }}
-                    selected={false}
-                    testID="create-new-configuration-from-plugin--dropdown-item"
-                  >
-                    <span>Create a new configuration</span>
-                  </Dropdown.Item>
-                  <Dropdown.Item
-                    key="Add-to-existing-configuration-telegraf-plugin"
-                    value="Add to an existing configuration"
-                    onClick={() => {}}
-                    selected={false}
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps & AddPluginToConfigurationCTAProps
+
+const AddPluginToConfigurationCTAComponent: FC<Props> = props => {
+  const {
+    contentID,
+    fetchTelegrafs,
+    history,
+    orgID,
+    pageContent,
+    telegrafs,
+    thumbnail,
+  } = props
+
+  useEffect(() => {
+    fetchTelegrafs()
```

Comment:
should this only `fetchTelegrafs` when the list of `telegrafs` changes? does this need to be fetched every time this component is rendered?

Reply:
This component is very high up on the hierarchy. It only renders once when a plugin's page is loaded. There may be re-renders but not re-fetches. The `useEffect` has an empty dependency list which means it only runs once when the component is mounted.

The beauty of this fetching is that on the plugin's page, the user can create a new telegraf configuration. But that gets added to state, which then is part of the resources checked by the sub-components. In other words, `telegrafs` remain updated after 1 fetch when the component is mounted, and there is no re-fetching even when a new telegraf is created on this page.